client/app/queue/CaseList/CaseListActions.js

```
@@ -90,14 +103,22 @@ export const fetchAppealsUsingVeteranId = (searchQuery) =>
       headers: { 'veteran-id': veteranId }
     }).
       then((response) => {
-        const returnedObject = JSON.parse(response.text);
+        let isResponseEmpty;
+        const returnedObject = (response.text) ? JSON.parse(response.text) : null;
 
-        if (_.size(returnedObject.appeals) === 0) {
+        if (returnedObject) {
+          isResponseEmpty = _.size(returnedObject.appeals) === 0 &&
+            _.size(returnedObject.claim_reviews) === 0;
+        }
+
+        if (!returnedObject || isResponseEmpty) {
```

I think we can simplify this logic. Since this is the first function of the then block, we know we're in the happy path and received a successful response from the server. This request hits `/appeals` which routes to `AppealsController.index` and there is only [one successful response we return](https://github.com/department-of-veterans-affairs/caseflow/pull/8098/files#diff-6c45b90fc8f3b5532860540b192918abR16) from that function so we can assume that is our response here. If we do that I think we can simplify this logic to something like:

```jsx
const returnedObject = JSON.parse(response.text);

if (_.size(returnedObject.appeals) === 0 && _.size(returnedObject.claim_reviews) === 0) {
  dispatch(fetchedNoAppealsUsingVeteranId(veteranId));

  return reject();
}
```

I'm not super experienced with Javascript so this may be incorrect though. Let me know what you think.

---

I don't love this either but I did encounter a condition where `response.text` is equal to `""`, and `JSON.parse("")` fails silently and exits, which led me to handle this condition specifically.